.github/workflows/release.yml

```diff
@@ -8,42 +8,63 @@ on:
 
 jobs:
 
+  build:
+    name: build distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
```

Comment:
Should specify Python 3.12 here explicitly. When Python 3.13 is released, there may be a short time before symengine wheels are available, and we don't want releases to fail in the meantime.

For this reason we really ought also to run this "build" step upon every push to `main`, as suggested in the pypa docs.

Reply:
Yes, those seem like reasonable ideas. Having this run on `main` should work if we loosen the run conditions at the top of this file but restrict the conditions for the second and third job in here.